pkg/controller/remotemachineset/remotemachineset_controller.go

```diff
@@ -341,10 +341,10 @@ func (r *ReconcileRemoteMachineSet) generateMachineSets(
 		ms.Spec.Template.Spec.Taints = pool.Spec.Taints
 
 		// Add the managed-by-Hive annotation:
-		if ms.Annotations == nil {
-			ms.Annotations = make(map[string]string, 1)
+		if ms.Labels == nil {
```

Comment:
This check was already done a few lines back. It does not need to be done again.